spec/models/spree/order_contents_spec.rb

```diff
@@ -223,47 +223,68 @@
         end
       end
     end
+  end
 
-    describe "#update_cart" do
-      subject { order_contents.update_cart(update_params) }
+  describe "#update_cart" do
+    subject { order_contents.update_cart(update_params) }
 
-      let(:update_params) do
-        {
-          line_items_attributes: { id: @line_item.id, quantity: quantity, options: {}}
-        }
+    let(:update_params) do
+      {
+        line_items_attributes: { id: @line_item.id, quantity: quantity, options: {}}
+      }
+    end
+
+    context "for a gift card line item" do
+      before do
+        variant.product.update_attributes(gift_card: true)
+        @line_item = order_contents.add(variant, 2, options)
       end
 
-      context "for a gift card line item" do
-        before do
-          variant.product.update_attributes(gift_card: true)
-          @line_item = order_contents.add(variant, 2, options)
+      context "line item is being updated to a higher quantity" do
+        let(:quantity) { "4" }
+
+        it "creates new gift cards" do
+          expect { subject }.to change { Spree::VirtualGiftCard.count }.by(2)
         end
+      end
 
-        context "line item is being updated to a higher quantity" do
-          let(:quantity) { "4" }
+      context "line item is being updated to a lower quantity" do
+        context "one lower" do
```

Comment:
I don't know if that's in the scope of this PR,
but this is strange, as when you create a gift card, you add some text, a recipient etc (so they all are potentially different...)
When you remove one out of the stack, which one is it ? and why ?

Also when working on something gift cards related a while ago, I noticed that some parts of the app were actively trying to limit the amount of gift cards in the cart to 1. Maybe this part was part of it too.
Nick confirmed to me at the time that it was intended for users to only ever have 1 gift card per order.

Reply:
I had similar questions about how this gem "works," but it seemed like I'd fall down an extremely deep and dark hole trying to shave this particular 🐃 . I opted instead to trust that everything the gem does works as we'd like it to, and limited my changes to supporting an array of line-item attributes.

Reply:
Ok, I was wondering if we really want to support multiple gift cards here, as it is an explicit requirement that there is only one gift card per cart.

Reply:
Do you think this consideration is in-scope?

Reply:
I don't know if it is in scope, but it seems to me that this override was made explicitly to prevent multiple gift cards in the cart, and fulfill the requirement no ?
Maybe I am missing something, but it seems to me that removing this, will break this requirement. Which may lead to unexpected results elsewhere.

Reply:
I wipped up a temp branch to test all is good, and all functionality is preserved, so all good here!